### PR TITLE
java-quickstart replace scenarios by DAML Script

### PR DESCRIPTION
**Pull request by aherrmann-da**
This replaces the scenarios in the java quickstart project by DAML script test cases.

The following changes were required
* Replace `getParty` by `allocateParty`.
* Replace `create` and `exercise` by `createCmd` and `exerciseCmd`.
* Replace the pattern of `submit ... do create ...; exercise ...`
  by `submit ... do createAndExerciseCmd ... ...`.
* Replace instances of `submit ... do fetch ...` by `query ... ...`.
  To that end I've added the following helper function.
  ```
  queryIou : Party -> ContractId Iou -> Optional Iou
  ```

I have tested these scripts in `daml studio` with `daml-sdk-head` as of c8b10091db6f69497ec2dca486bb0900f390fead.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml
Comment (cocreature) [inline docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml:9]: I think a better solution for this is to make a template that you can call via `createAndExercise` that provides a choice that calls `Fetch`. That provides a much more direct solution for translating arbitrary `Update` expressions passed to `submit` in scenarios to DAML Script.

**Review by cocreature — COMMENTED** [review]
LGTM if we replace the `query` stuff by a fetch in a choice. As mentioned elsewhere, I’d like to hold off on merging this until we have switched this on by default.
That is currently blocked on two things:
1. Documentation. I don’t think this needs much. Make https://docs.daml.com/daml/daml-studio.html#scenario-results a bit more general and mention DAML Script and a pointer from https://docs.daml.com/daml-script/index.html to the daml studio docs.
2. The codelens still says “Scenario results” instead of “Script results”.
2. Translate one of our internal examples to catch bugs. I’ll take care of that.

File: docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml
Comment (aherrmann-da) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml:9]: Initially I went with the `Fetch` approach. I thought it was a bit odd to have to define an extra template just for `fetch` when the functionality already exists in `query`, but I'm happy to change it back.

Comment (cocreature) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/QueryIou.daml:9]: The reason why I prefer the `Fetch` approach is that it works for arbitrary `Update` expressions, e.g., if you look at some of the monadic `Update` statements that you converted you got lucky that they were exactly `createAndExercise` but if they are something else, defining a choice for this works for everything whether it be `fetch`, monadic updates or anything else.

There is also a difference in that `query` does the filtering on the client side while `fetch` does it on the ledger.

File: docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml
Comment (cocreature) [inline docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml:14]: Perhaps worth adding a comment here that we do this because `fetch` is not a Ledger API command.

**Review by cocreature — APPROVED** [review]
Nice thank you! Once the source locations change has landed, I’m happy to merge this.

File: docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml
Comment (aherrmann) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml:18]: Good point, I've added a comment pointing to the relevant docs.daml.com as well.

File: rules_daml/daml.bzl
Comment (cocreature) [inline rules_daml/daml.bzl:335]: I’m not quite clear why you need the `-gt 0` stuff. Shouldn’t this do the right thing for zero deps as well?

**Review by cocreature — APPROVED** [review]
Thanks a lot for making my changes to `daml_test` work on Windows!

File: rules_daml/daml.bzl
Comment (aherrmann-da) [reply rules_daml/daml.bzl:335]: For zero deps it prints `,""` which then fails with
```
damlc: user error (Cannot resolve SDK dependency ''. Use daml assistant.)
```

File: docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml
Comment (hurryabit) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml:18]: Should we consider adding `fetch` to the ledger API? cc @gerolf-da

Comment (cocreature) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml:18]: We added `queryContractId` to the DAML Script API in the meantime and used it here.

Comment (gerolf-da) [reply docs/source/app-dev/bindings-java/quickstart/template-root/daml/Tests/Helper.daml:18]: @hurryabit: seems like an easy enough "surface area" change, but would need more spec on visibility, what other meta data would you want along with the LF Value itself, etc...

**Pull request merged**
